crates/task/src/lib.rs

```diff
@@ -1,8 +1,42 @@
 //! Task primitives for `HotShot`
 
+use async_broadcast::{SendError, Sender};
+#[cfg(async_executor_impl = "async-std")]
+use async_std::task::JoinHandle;
+#[cfg(async_executor_impl = "tokio")]
+use tokio::task::JoinHandle;
+
 /// Simple Dependency types
 pub mod dependency;
 /// Task which can uses dependencies
 pub mod dependency_task;
 /// Basic task types
 pub mod task;
+
+/// Cancel a task
+#[allow(clippy::unused_async)]
+pub async fn cancel_task<T>(task: JoinHandle<T>) {
+    #[cfg(async_executor_impl = "async-std")]
+    task.cancel().await;
+    #[cfg(async_executor_impl = "tokio")]
+    task.abort();
+}
+
+/// Helper function to send events and log errors
+pub async fn broadcast_event<E: Clone + std::fmt::Debug>(event: E, sender: &Sender<E>) {
```

Comment:
Why are these two functions moved from `task-impls` to here? Looks like they are still called within `task-impls` only.

Reply:
This is now also called in `types`